test/testServer.js

```diff
@@ -0,0 +1,12 @@
+var Hapi   = require("hapi")
+var routes = require("../lib/routes")
+
+var server = new Hapi.Server()
+
+server.connection({ host: "test" })
+server.route(routes)
+server.register(require("inject-then"), function (err) {
+  if (err) throw err
+})
+
+module.exports = server
```

Comment:
What if you made a `lib/server.js` that defined a function to build your server with this logic and added parameters to extend the server with new behavior, like:

```javascript
function init (port, plugins, next) {
    // provision a new server with a port and list of plugins, then execute a callback when it completes
}
```

You'd then have to call it with something like:

```javascript
require('./server').init(PORT, PLUGINS, function(error, server) {
  // next stuff
})
```

This way, you could keep most of the logic up to date without needing to sync up a test version of the server.